cli/README.md

```diff
@@ -132,6 +132,49 @@ The `usage` command allows you to list cluster share and usage for a particular
 In addition to totals, it displays a breakdown of usage by application and job group.
 As with `show`, `usage` will list usage across all configured clusters.
 
+### Plugins
+
+You can also specify custom functionality for some subcommands with your own plugin functions.
+You will need to add a `plugins` map to your `.cs.json` file. An example plugin config is shown below.
+
+```json
+{
+  "plugins": {
+    "plugin-name": {
+      "module-name": "module_name",
+      "path": "path_to_plugin_module",
+      "function-name": "function_name"
+    }
+  }
+}
+```
+
+Plugins are available for the following subcommands:
+
+#### `ls`
+
+The `ls` command has the plugin name `retrieve-job-instance-files`.
+The plugin function must take in an instance, sandbox directory, and path,
+and must return an array of dicts containing the file metadata fields
+`path`, `mode`, `nlink`, `size`, `uid`, `gid`, and `mtime`.
+
+#### `cat`
+
+The `cat` command has the plugin name `download-job-instance-file`.
+The plugin function must take in an instance, sandbox directory, and path,
+and must return an object with an `iter_content` method which generates the file contents.
```

Comment:
This feels a little awkward, since it uses a function from `Response` but doesn't actually require a `Response`. 

We could:

- Make the requirement stronger and say that the function must return a valid `requests` `Response` object.
- Instead of requiring `iter_content` at all (which is a `Response`-specific concept), just require that the function return any valid Python iterator. This might make it feel more like the other two plugin functions, which don't assume you're using a specific HTTP library.

What do you think?

Reply:
Having the function return a valid requests Response object probably makes more sense. The iterator approach will likely be more confusing since your iterator will still have to be wrapped in an `iter_content` method, unless we want to change the API of `read_file`

Reply:
I'm fine with changing the API of `read_file` if it makes this cleaner